.github/steps/2-step.md

```diff
@@ -0,0 +1,165 @@
+
+   <img width="250" alt="image" src="https://github.com/user-attachments/assets/4a2ff616-aedd-41b5-a24c-82014e98bbee" />
+
+1. Around line 6, simplify the `on` trigger. Remove the `push` trigger and keep the `pull_request` trigger.
+
+   ```yml
+   on:
+     pull_request:
+       branches: ["main"]
+   ```
+
+1. Around line 38, change the command to show more details in the logs.
+
+   ```yml
+   - name: Test with pytest
+     run: |
+       pytest --verbose
+   ```
+
+1. Above the editor, on the right, click the **Commit changes...** button. Commit directly to the `main` branch.
+
+<details>
+<summary>Having trouble? 🤷‍♂️</summary>
+
+Indention of `.yml` files is important. If you are getting syntax errors, that may be the reason.
+
+Finished workflow file: `.github/workflows/python-package.yml.example`
+
+<details>
+
+### ⌨️ Activity: Add a workflow to show test coverage
+
+1. Switch to the VS Code Codespace.
+
+1. Check the status bar for a pending update. Click it to pull your recently committed workflow.
+
+   <img width="130" alt="image" src="https://github.com/user-attachments/assets/c3eb17cf-a18e-4fae-bf18-cda900c7c6d3" />
+
+1. In the left navigation, select the **Explorer** tab to show the project files.
+
+1. Expand the `.github/workflows/` folder.
+
+1. Add a new file with the following name and open it.
+
+   ```txt
+   python-coverage.yml
+   ```
+
+1. Enter the name and set it to trigger on pull requests targeting the `main` branch.
+
+   ```yml
+   name: Python Coverage
+
+   on:
+     pull_request:
+       branches:
+         - main
+
+   permissions:
+     pull-requests: write
+   ```
+
+1. Add the `python-coverage` job and a first step that gets the repository content.
+
+   ```yml
+   jobs:
+     python-coverage:
+       runs-on: ubuntu-latest
+
+       steps:
+         - name: Checkout code
+           uses: actions/checkout@v4
+   ```
+
+1. Add steps to install Python and required packages.
+
+   ```yml
+   - name: Set up Python
+     uses: actions/setup-python@v4
+     with:
+       python-version: 3.13
+
+   - name: Install dependencies
+     run: |
+       pip install -r requirements.txt
+       pip install pytest==8.4.1
+       pip install coverage==7.9
+       pip install pytest-cov==6.2.1
+   ```
+
+1. Add steps to run the coverage report on the `src` folder.
+
+   ```yml
+   - name: Run tests and generate coverage details
+     run: pytest --cov=src
+   ```
+
+1. Add a final step that uses a pre-made GitHub Action to share the coverage report as a comment on the pull request.
+
+   {% raw %}
+
+   ```yml
+   - name: Coverage comment
+     uses: py-cov-action/python-coverage-comment-action@v3
+     with:
+       GITHUB_TOKEN: ${{ github.token }}
+       MINIMUM_GREEN: 90
+       MINIMUM_ORANGE: 70
+
+   - name: Fail if below threshold
+     run: coverage report --fail-under=90
+   ```
+
+   {% endraw %}
+
+1. Commit and push the changes to your `python-coverage.yml` file to the `main` branch.
+
+> [!TIP]
```

Comment:
GitHub syntax broke ordering and didn't display correctly

<img width="712" height="186" alt="image" src="https://github.com/user-attachments/assets/c091972f-de47-423d-9dc3-e8c9538a79c0" />